assets/js/editor-components/product-control/index.tsx

```diff
@@ -142,29 +165,37 @@ const ProductControl = ( {
 		} else if ( showVariations ) {
 			return renderItemWithVariations;
 		}
-		return null;
+		return () => {
+			return null;
+		};
 	};
 
 	if ( error ) {
 		return <ErrorMessage error={ error } />;
 	}
 
 	const currentVariations =
-		variations && variations[ expandedProduct ]
+		variations && expandedProduct !== null && variations[ expandedProduct ]
```

Comment:
This won't change behaviour, since indexing an object by null will return undefined.. unless it had a prop called `null`. 😰